src/mlpack/bindings/cli/in_place_copy.hpp

```diff
@@ -50,14 +50,34 @@ template<typename T>
 void InPlaceCopyInternal(
     util::ParamData& d,
     util::ParamData& input,
-    const typename std::enable_if<
-        arma::is_arma_type<T>::value ||
-        std::is_same<T,
-                     std::tuple<mlpack::data::DatasetInfo, arma::mat>>::value ||
-        data::HasSerialize<T>::value>::type* = 0)
+    const typename std::enable_if<arma::is_arma_type<T>::value ||
+                                  std::is_same<T,
+        std::tuple<mlpack::data::DatasetInfo, arma::mat>>::value>::type* = 0)
 {
   // Make the output filename the same as the input filename.
-  typedef std::tuple<T, typename ParameterType<T>::type> TupleType;
+  typedef std::tuple<T, typename ParameterType<T>::type, size_t, size_t> TupleType;
+  TupleType& tuple = *boost::any_cast<TupleType>(&d.value);
+  std::string& value = std::get<1>(tuple);
+
+  const TupleType& inputTuple = *boost::any_cast<TupleType>(&input.value);
+  value = std::get<1>(inputTuple);
```

Comment:
Although, I suppose there would be a little bit of extra adaptation because now getting the filename would be `std::get<0>(std::get<1>(tuple))`.  But, still, I think maybe that is a cleaner way to do it, since all of the auxiliary information that needs to be stored about these types of parameters are stored nicely in a `typename ParameterType<T>::type`. :+1: